l10n_th_base_location/models/res_country_state.py

```diff
@@ -0,0 +1,19 @@
+# Copyright 2021 Sansiri Tanachutiwat
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+
+from odoo import models
+
+
+class CountryState(models.Model):
+    _inherit = "res.country.state"
+
+    def name_get(self):
+        result = []
+        for record in self:
+            if record.country_id.code == "TH":
+                result.append((record.id, record.name))
+            else:
+                result.append(
```

Comment:
I think we should write it the way that it call back to super().name_get() for non "TH" records

Note: I understand from the code that most name_get() does not call super(), only some do. Nevertheless, we should try not to break it, in case, other country is doing the same as what we do. WDYT?

Reply:
For compability, it is better to revise to module as suggested.

However, the method will be much more complicated due to the list structure of `result` variables. Let's me think of a way to do it neatly.